Src/Fido2/AttestationFormat/AppleAppAttest.cs

```diff
@@ -0,0 +1,124 @@
+﻿using System;
+using System.Formats.Asn1;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using Fido2NetLib.Cbor;
+using Fido2NetLib.Objects;
+
+namespace Fido2NetLib
+{
+    internal sealed class AppleAppAttest : AttestationVerifier
+    {
+        public static byte[] GetAppleAppIdFromCredCertExtValue(X509ExtensionCollection exts)
+        {
+            var appleExtension = exts.Cast<X509Extension>().FirstOrDefault(static e => e.Oid!.Value is "1.2.840.113635.100.8.5");
+
+            if (appleExtension is null || appleExtension.RawData is null)
+                throw new Fido2VerificationException("Extension with OID 1.2.840.113635.100.8.5 not found on Apple AppAttest credCert");
+
+                var appleAttestationASN = Asn1Element.Decode(appleExtension.RawData);
+                appleAttestationASN.CheckTag(Asn1Tag.Sequence);
+                foreach (Asn1Element s in appleAttestationASN.Sequence)
+                {
+                    if (s.TagValue is 1204)
+                    {
+                        // App ID is the concatenation of your 10-digit team identifier, a period, and your app's CFBundleIdentifier value 
+                        s.CheckExactSequenceLength(1);
+                        s[0].CheckTag(Asn1Tag.PrimitiveOctetString);
+                        return s[0].GetOctetString();
+                    }
+                }
+            throw new Fido2VerificationException("Apple AppAttest attestation extension 1.2.840.113635.100.8.5 has invalid data");
+        }
+
+        // From https://www.apple.com/certificateauthority/Apple_App_Attestation_Root_CA.pem
+        internal static readonly string appleAppAttestationRootCA = "MIICITCCAaegAwIBAgIQC/O+DvHN0uD7jG5yH2IXmDAKBggqhkjOPQQDAzBSMSYwJAYDVQQDDB1BcHBsZSBBcHAgQXR0ZXN0YXRpb24gUm9vdCBDQTETMBEGA1UECgwKQXBwbGUgSW5jLjETMBEGA1UECAwKQ2FsaWZvcm5pYTAeFw0yMDAzMTgxODMyNTNaFw00NTAzMTUwMDAwMDBaMFIxJjAkBgNVBAMMHUFwcGxlIEFwcCBBdHRlc3RhdGlvbiBSb290IENBMRMwEQYDVQQKDApBcHBsZSBJbmMuMRMwEQYDVQQIDApDYWxpZm9ybmlhMHYwEAYHKoZIzj0CAQYFK4EEACIDYgAERTHhmLW07ATaFQIEVwTtT4dyctdhNbJhFs/Ii2FdCgAHGbpphY3+d8qjuDngIN3WVhQUBHAoMeQ/cLiP1sOUtgjqK9auYen1mMEvRq9Sk3Jm5X8U62H+xTD3FE9TgS41o0IwQDAPBgNVHRMBAf8EBTADAQH/MB0GA1UdDgQWBBSskRBTM72+aEH/pwyp5frq5eWKoTAOBgNVHQ8BAf8EBAMCAQYwCgYIKoZIzj0EAwMDaAAwZQIwQgFGnByvsiVbpTKwSga0kP0e8EeDS4+sQmTvb7vn53O5+FRXgeLhpJ06ysC5PrOyAjEAp5U4xDgEgllF7En3VcE3iexZZtKeYnpqtijVoyFraWVIyd/dganmrduC1bmTBGwD";
```

Comment:
maybe get the RootCA from the url (and use a hardwired one as a fallback?)
It's unlikely, but I am thinking of the case where it's revoked.

Reply:
Hard coding the root is the best practice. You can't revoke a root CA certificate.